frontend/src/components/common/rich-text-editor/controls/InlineAttachmentControl.tsx

```diff
@@ -0,0 +1,179 @@
+import { EditorState, Modifier, SelectionState } from 'draft-js'
+import { useContext, useState } from 'react'
+
+import type { FormEvent, MouseEvent as ReactMouseEvent } from 'react'
+
+import Dropzone from 'react-dropzone'
+
+import { EditorContext } from '../RichTextEditor'
+import styles from '../RichTextEditor.module.scss'
+
+import CloseButton from 'components/common/close-button'
+import { uploadCommonCampaignAttachment } from 'services/attachment.service'
+
+const InlineAttachmentForm = ({
+  onChange,
+  doCollapse,
+}: {
+  onChange: (key: string, ...vals: string[]) => void
+  doCollapse: () => void
+}) => {
+  const [text, setText] = useState('')
+  const [link, setLink] = useState('')
+  const [error, setError] = useState('')
+  const [isLoading, setIsLoading] = useState(false)
+
+  function stopPropagation(e: ReactMouseEvent<HTMLElement>) {
+    e.stopPropagation()
+  }
+
+  async function handleDrop(files: Array<File>) {
+    setIsLoading(true)
+    if (!files || files.length === 0) return
+    try {
+      setLink(await uploadCommonCampaignAttachment(files[0]))
+      setText(files[0].name)
+    } catch (e) {
+      setError('Failed to upload image. Please check file size to be < 50MB')
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  function handleSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault()
+    onChange(link, text)
+  }
+  return (
+    <form
+      onClick={stopPropagation}
+      onSubmit={handleSubmit}
+      className={styles.form}
+    >
+      <div className={styles.top}>
+        <div>
+          <p className={styles.header}>Embed attachment</p>
+        </div>
+        <CloseButton onClick={doCollapse} className={styles.closeButton} />
+      </div>
+      {!isLoading && !link && (
+        <div className={styles.item}>
+          <Dropzone
+            maxFiles={1}
```

Comment:
[suggestion] i understand that this is current max files of 1, but in future if we intend to increase max files / lower the max size per file etc, this might be prone to errors. we can consider using static constants for max files and size